### PR TITLE
1501 info entity context

### PR DESCRIPTION
**Pull request by ramtob**

### CODE REVIEW

File: src/app/@ansyn/ansyn/app-effects/effects/map/visualizers.app.effects.ts [hunk 1]
```diff
@@ -10,15 +9,14 @@ import { ImageryCommunicatorService } from '@ansyn/imagery/communicator-service/
...
+	SetPinLocationModeAction,
 	ShowOverlaysFootprintAction,
-	StartMouseShadow,
-	StopMouseShadow,
 	ToolsActionsTypes
 } from '@ansyn/menu-items/tools/actions/tools.actions';
 import { MapFacadeService } from '@ansyn/map-facade/services/map-facade.service';
 import { IMapState, mapStateSelector } from '@ansyn/map-facade/reducers/map.reducer';
-import { ContextEntityVisualizer } from '../../../app-providers/app-visualizers/context-entity.visualizer';
+import { ContextEntityVisualizer } from '../../../../plugins/openlayers/visualizers/contexts/context-entity.visualizer';
```
Comment (yairtawil) [inline src/app/@ansyn/ansyn/app-effects/effects/map/visualizers.app.effects.ts]: fix

File: src/app/@ansyn/ansyn/app-effects/effects/cases/select-case.app.effects.ts [hunk 2]
```diff
@@ -50,7 +51,7 @@ export class SelectCaseAppEffects {
 		// map
 		const { data, activeMapId } = state.maps;
 		// core
```
Comment (yairtawil) [inline src/app/@ansyn/ansyn/app-effects/effects/cases/select-case.app.effects.ts]: comment to context

File: src/app/@ansyn/ansyn/app-providers/overlay-source-providers/planet/planet-source-provider.ts [hunk 3]
```diff
@@ -138,43 +138,27 @@ export class PlanetSourceProvider extends BaseOverlaySourceProvider {
...
-
-						const date = new Date(overlay.photoTime);
-						if (date < startDate) {
-							startDate = date;
-						}
-						if (endDate < date) {
-							endDate = date;
-						}
-					});
+					const overlaysDates = overlays.map(overlay => moment(overlay.date));
```
Comment (yairtawil) [inline src/app/@ansyn/ansyn/app-providers/overlay-source-providers/planet/planet-source-provider.ts]: 👑

File: src/app/@ansyn/ansyn/app-providers/overlay-source-providers/planet/planet-source-provider.ts [hunk 4]
```diff
@@ -204,8 +196,60 @@ export class PlanetSourceProvider extends BaseOverlaySourceProvider {
...
+					endDate = moment.max(overlaysDates).toDate();
+				}
+				return endDate;
+			})
+			.catch((error: HttpResponseBase | any) => {
+				return this.errorHandlerService.httpErrorHandle(error);
+			});
+
+		return startDate$
```
Comment (yairtawil) [inline src/app/@ansyn/ansyn/app-providers/overlay-source-providers/planet/planet-source-provider.ts]: try forkJoin

File: src/app/@ansyn/context/effects/context.effects.ts [hunk 5]
```diff
@@ -16,24 +19,26 @@ import { StartAndEndDate } from '@ansyn/overlays/models/base-overlay-source-prov
...
-		.map(([action, contexts]: [LoadDefaultCaseAction, any[]]) => {
+		.withLatestFrom(this.store.select(selectContextsArray), this.store.select(selectContextsParams))
+		.map(([action, contexts, params]: [LoadDefaultCaseAction, any[], ContextParams]) => {
 			const context = contexts.find(({ id }) => action.payload.context === id);
-			return [action, context];
+			return [action, context, params];
 		});
 
-	setContext = mergeMap(([action, context]: [LoadDefaultCaseAction, any]) => {
+	setContext = mergeMap(([action, context, contextParams]: [LoadDefaultCaseAction, any, ContextParams]) => {
```
Comment (yairtawil) [inline src/app/@ansyn/context/effects/context.effects.ts]: Context instead of any

File: src/app/@ansyn/context/effects/context.effects.ts [hunk 6]
```diff
@@ -60,7 +65,7 @@ export class ContextEffects {
 	@Effect()
 	loadExistingDefaultCaseContext$: Observable<SetContextParamsAction | SelectCaseAction> =
 		this.loadDefaultCaseContext$
-			.filter(([action, context]: [LoadDefaultCaseAction, any]) => Boolean(context))
+			.filter(([action, context]: [LoadDefaultCaseAction, any, ContextParams]) => Boolean(context))
```
Comment (yairtawil) [inline src/app/@ansyn/context/effects/context.effects.ts]: 🔝

File: src/app/@ansyn/context/effects/context.effects.ts [hunk 7]
```diff
@@ -73,14 +78,15 @@ export class ContextEffects {
 	@Effect()
 	loadNotExistingDefaultCaseContext$: Observable<any> =
 		this.loadDefaultCaseContext$
-			.filter(([action, context]: [LoadDefaultCaseAction, any]) => !(Boolean(context)))
-			.mergeMap(([action, context]: [LoadDefaultCaseAction, any]) => {
+			.filter(([action, context]: [LoadDefaultCaseAction, any, ContextParams]) => !(Boolean(context)))
+			.mergeMap(([action, context, params]: [LoadDefaultCaseAction, any, ContextParams]) => {
```
Comment (yairtawil) [inline src/app/@ansyn/context/effects/context.effects.ts]: ⏫

File: src/app/@ansyn/context/effects/context.effects.ts [hunk 8]
```diff
@@ -129,6 +135,20 @@ export class ContextEffects {
 		.switchMap((action: LoadCaseAction) => this.casesService.loadCase(action.payload))
 		.map((dilutedCase) => new SelectDilutedCaseAction(dilutedCase));
 
+	@Effect()
+	setSpecialObjectsFromSelectedCase$: Observable<any> = this.actions$
```
Comment (yairtawil) [inline src/app/@ansyn/context/effects/context.effects.ts]: `select-case.app.effect`

File: src/app/@ansyn/ansyn/app-effects/effects/map/visualizers.app.effects.ts (hunk 1 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/map/visualizers.app.effects.ts]: done

File: src/app/@ansyn/ansyn/app-effects/effects/cases/select-case.app.effects.ts (hunk 2 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/cases/select-case.app.effects.ts]: done

File: src/app/@ansyn/ansyn/app-providers/overlay-source-providers/planet/planet-source-provider.ts (hunk 4 above)
Comment (ramtob) [reply src/app/@ansyn/ansyn/app-providers/overlay-source-providers/planet/planet-source-provider.ts]: done

File: src/app/@ansyn/context/effects/context.effects.ts (hunk 5 above)
Comment (ramtob) [reply src/app/@ansyn/context/effects/context.effects.ts]: done

File: src/app/@ansyn/context/effects/context.effects.ts (hunk 6 above)
Comment (ramtob) [reply src/app/@ansyn/context/effects/context.effects.ts]: 🔝

File: src/app/@ansyn/ansyn/app-effects/effects/cases/select-case.app.effects.ts [hunk 9]
```diff
@@ -39,6 +40,20 @@ export class SelectCaseAppEffects {
 		.ofType<SelectCaseAction>(CasesActionTypes.SELECT_CASE)
 		.mergeMap(({ payload }: SelectCaseAction) => this.selectCaseActions(payload));
 
+	@Effect()
+	setSpecialObjectsFromSelectedCase$: Observable<any> = this.actions$
```
Comment (yairtawil) [inline src/app/@ansyn/ansyn/app-effects/effects/cases/select-case.app.effects.ts]: remove setSpecialObjectsFromSelectedCase$ add functionality to selectCase$

Comment (ramtob) [reply src/app/@ansyn/ansyn/app-effects/effects/cases/select-case.app.effects.ts]: done

**Pull request merged**